### PR TITLE
Improvement

### PR DESCRIPTION
**Pull request by christy313**
## Summary
Improved the issue of refreshing the browser, the todo-list disappeared.

## what did I do?
- Saved data in local storage after inserting a new task

### CODE REVIEW

**Comment by christy313** [comment]
Hi Jessie,

This is a todo-list made with HTML, CSS, and vanilla JavaScript. and it is stored in local storage.
There are add/delete/check todo functions, the improvement is to press enter key down to add todo instead of pressing an add button.

I haven't done the edit todo part.

Christy

File: Script.js [hunk 1]
```diff
@@ -1,39 +1,73 @@
...
-    return;
+  if (content.trim() && e.key === "Enter") {
+    const todo = { id: `${new Date().getTime()}`, content, completed: false };
+    todos.push(todo);
+    localStorage.setItem("todos", JSON.stringify(todos));
+    addTodo(todo);
   }
+});
+
+const addTodo = (todo) => {
```
Comment (iamseye) [inline Script.js:26]: naming, ex: `createTaskInnerHTML`

File: Script.js [hunk 2]
```diff
@@ -1,39 +1,73 @@
...
+      todo.completed ? "checked" : ""
+    }>
+    <div class="todo-content">${escapeHtml(content)}</div>
+    <button class="delete">x</button>
+  `;
 
-  if (target.classList.contains("check")) {
-    const chk = target.parentNode.classList;
-    target.checked ? chk.add("done") : chk.remove("done");
+  todosContainer.appendChild(todoDiv);
```
Comment (iamseye) [inline Script.js:40]: `todosContainer` is a global variable, can you return todoDiv and append outside of the function?

File: Script.js [hunk 3]
```diff
@@ -1,39 +1,73 @@
...
-document.querySelector(".content").addEventListener("click", (e) => {
-  const { target } = e;
-  if (target.classList.contains("delete")) {
-    target.parentNode.remove();
-    return;
+  if (content.trim() && e.key === "Enter") {
+    const todo = { id: `${new Date().getTime()}`, content, completed: false };
+    todos.push(todo);
+    localStorage.setItem("todos", JSON.stringify(todos));
+    addTodo(todo);
```
Comment (iamseye) [inline Script.js:22]: `const newTaskInnerHTML = addTodo(todo)`
`todosContainer.appendChild(newTaskInnerHTML)`

**Review by iamseye — CHANGES_REQUESTED** [review]

File: index.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { createId, createTaskInnerHTML, completeTodo } from "./helper.js";
+
+let taskItems = JSON.parse(localStorage.getItem("taskItems")) || [];
```
Comment (iamseye) [inline index.js:3]: can you rename it? `savedTaskitems`?

File: index.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+// add task
+taskInput.addEventListener("keypress", (e) => {
+  const content = taskInput.value;
+
+  if (content.trim() && e.key === "Enter") {
+    const key = createId();
+    const taskValue = { content, completed: false };
+    const taskContent = [key, taskValue];
+    taskItems.push(taskContent);
```
Comment (iamseye) [inline index.js:26]: why do you have to update the savedTaksItems?

File: index.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    localStorage.setItem("taskItems", JSON.stringify(taskItems));
+    e.target.closest("div").remove();
+  }
+});
+
+// complete task
+tasksContainer.addEventListener("click", (e) => {
+  const checkBox = e.target;
+  const taskId = checkBox.parentNode.getAttribute("id");
+  completeTodo(taskId, checkBox, taskItems);
```
Comment (iamseye) [inline index.js:49]: This will be `checkTask` / `toggleTask` or can only `completeTask`  ?

File: Style.css [hunk 7]
```diff
@@ -1,5 +1,6 @@
 * {
```
Comment (iamseye) [inline Style.css:1]: The file naming `style.css` for syncing with other file names

File: index.js (hunk 5 above)
Comment (christy313) [reply index.js]: it needs to be pushed inside `savedTaskItems` to save in local storage

File: index.js (hunk 6 above)
Comment (christy313) [reply index.js]: I changed the function name from `completeTodo` to `checkTodo` for check/uncheck a checkbox

**Pull request merged**